lib/puppet/provider/package/pip.rb

```diff
@@ -29,7 +29,11 @@ def self.parse(line)
   # that's managed by `pip` or an empty array if `pip` is not available.
   def self.instances
     packages = []
-    pip_cmd = which(cmd) or return []
+    if cmd.class == Array
+      pip_cmd = ( which(cmd.first) or which(cmd.last) ) or return []
+    else
+      pip_cmd = which(cmd) or return []
```

Comment:
Same comment as above with respect to `or`.  If `cmd` is always an array, there's no need for this else anyway.